crossbuilder

```diff
@@ -845,10 +958,79 @@ if ! which lxd > /dev/null ; then
     fi
 fi
 
-check_lxd_accessible
 ensure_lxd_subuid
+check_lxd_accessible
+
+if [ -z $HOST_ARCH ] ; then
+    if which dpkg > /dev/null 2>&1 ; then
+        HOST_ARCH=`dpkg --print-architecture`
+    else
+        # To find out the host's debian architecture name, first find out debian multiarch tuple and use conversion table.
+        # Source: https://wiki.debian.org/Multiarch/Tuples
+
+        if [ -z $HOST_FARCH ] && which gcc > /dev/null 2>&1 ; then
+            HOST_FARCH=$(gcc -print-multiarch)
+
+            # On manjaro the 'gcc -print-multiarch' returns nothing.
+            # Use the 'gcc -dumpmachine' multiarch triplet result and strip the vendor part if the result contains it.
+            [ -z $HOST_FARCH ] && HOST_FARCH=$(gcc -dumpmachine | sed 's/^\([^-]\+\)-\([^-]\+\)-\([^-]\+\)-\([^-]\+\)$/\1-\3-\4/')
+        fi
+        #echo "HOST_FARCH=\"${HOST_FARCH}\""
+
+        declare -A MULTIARCH_TO_DEBIAN=( \
+            [alpha-linux-gnu]=alpha \
```

Comment:
please remove all the exotic architectures :-)

alpha, arc, mips, i386 (I bet no one has it running on the host), kfreebsd, powerpc...

We can always add them later if someone asks.

Also, please move all this code into a detect_host_architecure() function.

Reply:
Code moved to function & removed exotic architectures.
Please check, if some architecture needs to be removed or if I removed something that should not be removed.